src/main/java/org/spongepowered/api/data/manipulator/mutable/common/AbstractData.java

```diff
@@ -207,9 +207,17 @@ public boolean equals(Object obj) {
     }
 
     @Override
-    public DataContainer toContainer() {
-        return DataContainer.createNew()
-                .set(Queries.CONTENT_VERSION, getContentVersion());
+    public final DataContainer toContainer() {
+        return this.fillContainer(DataContainer.createNew().set(Queries.CONTENT_VERSION, getContentVersion()));
     }
 
+    /**
+     * Implement this method to add the data to be persisted.
+     *
+     * @param dataContainer The DataContainer
+     *
+     * @return The filled DataContainer
+     */
+    public abstract DataContainer fillContainer(DataContainer dataContainer);
```

Comment:
This should be made `protected` since it shouldn't need to by used outside this class structure.